src/tools/rbd/action/Bench.cc

```diff
@@ -293,18 +336,36 @@ int execute(const po::variables_map &vm) {
     return r;
   }
 
-  r = do_bench_write(image, bench_io_size, bench_io_threads, bench_bytes,
-                     bench_random);
+  r = do_bench(image, bench_io_type, bench_io_size, bench_io_threads,
+		     bench_bytes, bench_random);
   if (r < 0) {
-    std::cerr << "bench-write failed: " << cpp_strerror(r) << std::endl;
+    std::cerr << "do_bench failed: " << cpp_strerror(r) << std::endl;
     return r;
   }
   return 0;
 }
 
-Shell::Action action(
-  {"bench-write"}, {}, "Simple write benchmark.", "", &get_arguments, &execute);
+int execute_for_write(const po::variables_map &vm) {
+  return bench_execute(vm, BENCH_WRITE);
+}
+
+int execute_for_bench(const po::variables_map &vm) {
+  io_type_t bench_io_type;
+  if (vm.count("io-type")) {
+    bench_io_type = vm["io-type"].as<io_type_t>();
+  } else {
+    bench_io_type = BENCH_READ;
```

Comment:
I like `rbd bench read|write img` a little more than `rbd bench img --io-type read|write`, especially if we want `io-type` option to be non-optional, though I don't have a strong opinion, I would be fine with any way.